src/platform/packages/shared/kbn-unified-field-list/src/components/field_popover/field_popover_header.test.tsx

```diff
@@ -45,42 +46,29 @@ describe('UnifiedFieldList <FieldPopoverHeader />', () => {
       />
     );
 
-    expect(wrapper.text()).toBe(fieldName);
-    expect(
-      wrapper.find(`[data-test-subj="fieldPopoverHeader_addBreakdownField-${fieldName}"]`).exists()
-    ).toBeTruthy();
-    expect(
-      wrapper.find(`[data-test-subj="fieldPopoverHeader_addField-${fieldName}"]`).exists()
-    ).toBeTruthy();
-    expect(
-      wrapper.find(`[data-test-subj="fieldPopoverHeader_addExistsFilter-${fieldName}"]`).exists()
-    ).toBeTruthy();
-    expect(
-      wrapper.find(`[data-test-subj="fieldPopoverHeader_editField-${fieldName}"]`).exists()
-    ).toBeTruthy();
-    expect(
-      wrapper.find(`[data-test-subj="fieldPopoverHeader_deleteField-${fieldName}"]`).exists()
-    ).toBeTruthy();
-    expect(wrapper.find(EuiButtonIcon)).toHaveLength(5);
+    expect(await screen.findByText(fieldName)).toBeVisible();
+    expect(screen.queryByTestId(`fieldPopoverHeader_addBreakdownField-${fieldName}`)).toBeVisible();
+    expect(screen.queryByTestId(`fieldPopoverHeader_addField-${fieldName}`)).toBeVisible();
+    expect(screen.queryByTestId(`fieldPopoverHeader_addExistsFilter-${fieldName}`)).toBeVisible();
+    expect(screen.queryByTestId(`fieldPopoverHeader_editField-${fieldName}`)).toBeVisible();
+    expect(screen.queryByTestId(`fieldPopoverHeader_deleteField-${fieldName}`)).toBeVisible();
+    expect(screen.getAllByRole('button')).toHaveLength(5);
```

Comment:
the recommendation here would be to use `queryBy` when you want to assert that something isn't there, `getBy` when something is already there and `findBy` when something will be there

```suggestion
    expect(screen.getByText(fieldName)).toBeVisible();
    expect(screen.getByTestId(`fieldPopoverHeader_addBreakdownField-${fieldName}`)).toBeVisible();
    expect(screen.getByTestId(`fieldPopoverHeader_addField-${fieldName}`)).toBeVisible();
    expect(screen.getByTestId(`fieldPopoverHeader_addExistsFilter-${fieldName}`)).toBeVisible();
    expect(screen.getByTestId(`fieldPopoverHeader_editField-${fieldName}`)).toBeVisible();
    expect(screen.getByTestId(`fieldPopoverHeader_deleteField-${fieldName}`)).toBeVisible();
    expect(screen.getAllByRole('button')).toHaveLength(5);
```